log4j-slf4j-impl/src/main/java/org/apache/logging/slf4j/Log4jLogger.java

```diff
@@ -411,13 +411,17 @@ private void writeObject(final ObjectOutputStream aOutputStream) throws IOExcept
         aOutputStream.defaultWriteObject();
     }
 
+    private static Boolean eventDataLoaded = null;
     private static EventDataConverter createConverter() {
-        try {
-            LoaderUtil.loadClass("org.slf4j.ext.EventData");
-            return new EventDataConverter();
-        } catch (final ClassNotFoundException cnfe) {
-            return null;
+        if (eventDataLoaded == null) {
+            try {
+                LoaderUtil.loadClass("org.slf4j.ext.EventData");
+                eventDataLoaded = true;
+            } catch (final ClassNotFoundException cnfe) {
+                eventDataLoaded = false;
+            }
         }
+        return eventDataLoaded ? new EventDataConverter() : null;
```

Comment:
Instead of modifying this method, what if we update the `converter` field to be static and call this method on class initialization? That way instead of requiring work (and allocation!) every time a logger is created, it's done once when the application loads.

Reply:
done

Reply:
Great, thanks!